src/Microsoft.OData.Core/ServiceCollectionExtensions.cs

```diff
@@ -0,0 +1,62 @@
+﻿//---------------------------------------------------------------------
+// <copyright file="ServiceCollectionExtensions.cs" company="Microsoft">
+// Copyright (C) Microsoft Corporation. All rights reserved. See License.txt in the project root for license information.
+// </copyright>
+//---------------------------------------------------------------------
+
+using System;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.OData.Edm;
+using Microsoft.OData.Json;
+using Microsoft.OData.UriParser;
+
+namespace Microsoft.OData.Core

```

Comment:
These extensions should actually live in `Microsoft.Extensions.DependencyInjection` so that developers don't have to add `using Microsoft.OData.Core` in the file, and the extensions magically appear to them.

Best practices are to ALWAYS include extension methods in the namespace they are extending. Making a developer futz around for the right namespace for an extension they are expecting is _no bueno_.

Reply:
updated